HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,736 @@
+
+Below are a some example scenarios to highlight how the tranaction record will expose state changes across `TokenAirdrop`, `TokenClaim` and `TokenCancel`.
+
+Scenario 1: If Alice airdrops with the following transfer breakout
+
+- 5 FTs of token T1 to Bob, where Bob is already associated with T1
+- NFT serial 1 & 2 of T2 to Bob who is associated with token T2
+- 10 FTs of token T3 to Carol, where Carol is not associated with T3 but has available association slots
+- NFT serials 3 & 4 of T4 to Carol, where Carol is not associated with T4 but has available association slots
+- 15 FTs of token T5 to Dave, where Dave is not associated with T5 and has no available association slots
+- NFT serial 5 & 6 of T6 to Dave, where Dave is not associated with T6 and has no available association slots
+- 20 FTs of token T7 to Eric, where Eric is not associated with T7 and has no available association slots
+- NFT serials 7 & 8 of T8 to Eric, where Eric is not associated with T8 and has no available association slots
+
+The record file emitted from a consensus node will contain
+
+`tokenTransferLists` containing 
+
+- Alice debit of 5 T1 tokens → Bob  credit of 5 T1 tokens
+- Alice debit of NFT serials 1 & 2 of token T2 → Bob credit of NFT serials 1 & 2 of token T2
+- Alice debit of 10 T3 tokens → Carol  credit of 10 T3 tokens
+- Alice debit of NFT serials 3 & 4 of token T4 → Carol credit of NFT serials 1 & 2 of token T4
+
+`automatic_token_associations` containing
+
+- Carols automatic association with token T3
+- Carols automatic association with token T4
+
+`pending_airdrops` containing
+
+- Alices intent to transfer 15 T5 tokens to Dave
+- Alices intent to transfer NFT serials 5 & 6 of token T6 to Dave
+
+Note: In the transaction body Alice had split her intended transfers to Dave in two but in the result exposed by the record they were combined. TokenAirdrops may be combined in an additive fashion. It is not possible to subtract. For this a sender may cancel and reissue their airdrop 
+
+Scenario 2: If Dave submits a TokenClaim for token T5 and a Token claim of serial 5 of T6
+
+`tokenTransferLists`
+
+- Alice debit of 15 T5 tokens → Dave credit of 15 T5 tokens
+- Alice debit of NFT serial 5 of token T6 → Dave credit of NFT serial 5 of token T6
+
+`automatic_token_associations` containing
+
+- Daves automatic association with token T5
+- Dave automatic association with token T6
+
+`pending_airdrops` will be empty
+
+Note: NFT serial 6 of T6 is left in pending airdrop, but all of the pending fungible amount of T5 was transferred 
+
+Scenario 3: Eric submits a TokenCancelAirdrop for token T7 and a TokenCancelAirdrop of serial 7 of T8
+
+`tokenTransferLists` will be empty
+
+`automatic_token_associations` will be empty
+
+`pending_airdrops` will be empty, details are in transaction body
```

Comment:
Shouldn't we have

- Alices intent to transfer 15 T5 tokens to Dave
- Alices intent to transfer NFT serials 5 & 6 of token T6 to Dave
- Alices intent to transfer NFT serials 8 of token T8 to Eric

Reply:
We considered it but it would be a repetition of information already in the transaction body, so to reduce the size the byte size on the network and mirror node we opt to leave it out